server-side has() can only check for custom perms (#1836)
clerk/clerk-docs: docs/references/backend/types/auth-object.mdx

description: The Auth object contains information about the current user's session.
---

The `Auth` object contains important information like the current user's session ID, user ID, and organization ID. It also contains methods to check for permissions and retrieve the current user's session token.

## How to access the `Auth` object

> The `Auth` object is available on the `request` object in server contexts. Some frameworks provide a helper that returns the `Auth` object; see the following table for more information.

Reviewer: ```suggestion
The `Auth` object is available on the `request` object in server contexts. Some frameworks provide a helper that returns the `Auth` object. See the following table for more information.
```

The `orgPermissions` property on the `Auth` object has the type `OrganizationCustomPermissionKey`.

> `OrganizationCustomPermissionKey` is a string that represents the permission of the user in the organization. Clerk provides [default system permissions](/docs/organizations/roles-permissions#system-permissions), and you can create [custom permissions](/docs/organizations/create-roles-permissions#create-a-new-permission-for-your-organization).

Reviewer: ```suggestion
`OrganizationCustomPermissionKey` is a string representing a user's permission in an organization. Clerk provides [default system permissions](/docs/organizations/roles-permissions#system-permissions) and allows you to create [custom permissions](/docs/organizations/create-roles-permissions#create-a-new-permission-for-your-organization).
```

  The age of the factor level to check for. Value should be greater than or equal to 1 and less than 99,999.
</Properties>

#### `has()` authorization example

You can use `has()` to check if a user is authorized to access a component.

<Include src="_partials/has-warning" />

> In the following example, `has()` is used to check if the user has the `org:team_settings:manage` permission. If the user does not have the permission, `null` is returned and the page is not rendered. This example is written for Next.js App Router, but it can be adapted to other frameworks by using the appropriate method for accessing the `Auth` object.

Reviewer: ```suggestion
In the following example, `has()` is used to check if the user has the `org:team_settings:manage` permission. If the user doesn't have the permission, `null` is returned and the page isn't rendered. This example is written for Next.js App Router, but can be adapted to other frameworks by using the appropriate method for accessing the `Auth` object.
```
